File: go/0035_search_insert_position.go
package main

func searchInsert(nums []int, target int) int {
	nlen := len(nums)

	left, right := 0, nlen-1
	for right-left > 0 {
		mid := (left + right) / 2
		if target < nums[mid] {
			right = mid - 1
		} else if target > nums[mid] {
			left = mid + 1
		} else {
			return mid
		}
	}

	if target <= nums[left] {
		return left
	}
	return left + 1
}

File: python/0035_search_insert_position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        nlen = len(nums)
        if nlen == 0:
            return 0

        left = -1
        right = nlen

        while right - left > 1:
            mid = left + (right - left) // 2
            if nums[mid] < target:
                left = mid
            else:
                right = mid
        
        return right